### PR TITLE
Make `meta` optional (default to `undefined`)

### PR DESCRIPTION
**Pull request by wub**

### CODE REVIEW

**Comment by wub** [comment]
Sorry, not an expert in versioning - what should this be bumped to?

**Comment by unional** [comment]
I'm ok with it.

One issue remain is this: https://github.com/Microsoft/TypeScript/issues/12400#issuecomment-296360175

That's why I didn't add the default generics yet.

You can see the behavior here:
http://www.typescriptlang.org/play/index.html#src=interface%20FSA%3CP%2C%20M%20%3D%20undefined%3E%20%7B%0D%0A%20%20%20%20payload%3A%20P%0D%0A%20%20%20%20meta%3A%20M%0D%0A%7D%0D%0A%0D%0Alet%20x%3A%20FSA%3Cnumber%3E%20%3D%20%7B%20payload%3A%201%20%7D%0D%0A%0D%0Ax%20%3D%20%7B%20payload%3A%201%2C%20meta%3A%20undefined%20%7D%0D%0A%0D%0Alet%20y%20%3D%20%7B%20payload%3A%201%20%7D%20as%20FSA%3Cnumber%3E%0D%0A

You can see type declaration (`x`) is not working. Type casting (`y`) is working thou.

File: src/index.d.ts [hunk 1]
```diff
@@ -26,26 +26,26 @@ export interface FluxStandardAction<Payload, Meta> {
   meta: Meta;
```
Comment (leesiongchan) [inline src/index.d.ts]: Isn't this supposed to be optional?

Comment (unional) [reply src/index.d.ts]: Nope. That's the whole point. It should not be optional. See #53 and related discussion.

Comment (rhys-vdw) [reply src/index.d.ts]: Hey, sorry to butt in here. Just hit this issue and am a bit confused... Currently I have this issue:

```ts
const x: FSA<string, undefined> = { type: "foo", payload: "hello" }
```
```
Type '{ type: string; payload: string; }' is not assignable to type 'FluxStandardAction<string, undefined>'.
  Property 'meta' is missing in type '{ type: string; payload: string; }'.
```

So setting the type to `undefined` does not make the attribute optional.

I think the DefinitelyTyped defs handle this [in a more appropriate way](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/7f0c54c13065b2a7f8e26a9fcc60e8b738916a77/types/flux-standard-action/index.d.ts#L18).

Comment (unional) [reply src/index.d.ts]: We discussed that here: https://github.com/acdlite/flux-standard-action/pull/53#issuecomment-279258673

It is debatable on which is a better solution.
I would say when https://github.com/Microsoft/TypeScript/issues/12400#issuecomment-296360175 is resolved, we will be just fine.

Comment (rhys-vdw) [reply src/index.d.ts]: Ah yeah, makes sense. It's just annoying having to add extra code to satisfy types, but a small price to pay I suppose. :)

**Comment by JaKXz** [comment]
Okay so unless anyone is opposed I'm going to merge this and release it as a patch today :)

**Comment by JaKXz** [comment]
`+ flux-standard-action@2.0.1` has been published! sorry for the delay, it completely slipped my mind before.

In related news, #100 would be a prime candidate for another patch like this one...

**Pull request merged**